pcdsdevices/lens.py

```diff
@@ -80,17 +81,61 @@ def __init__(self, prefix, *, name, **kwargs):
 
 
 class LensStackBase(BaseInterface, PseudoPositioner):
-    """Class for Be lens macros and safe operations."""
+    """
+    Class for Be lens macros and safe operations.
+
+    x, y, z: real motors that move the position of the lens set
+    calib_z, beam_size pseudo motors
+
+    An ophyd PseudoPositioner relates one or more pseudo (virtual) axes to one
+    or more real (physical) axes via forward and inverse calculations.
+
+    The purpose of this class is using the motor that moves the z stage of
+    BeLensStack so that we can scan the focal size. What needs to be done is
+    move the x motor and then the combination motor that when the z motor moves
+    we also move the x and y to compensate for the stage not being perfect.
+
+    What do we usually do:
+    For each lens pack that is in, tweak x and y until the fixture on YAG
+    screen is as pretty as it can be - save this position.
+    Move the z motor on one extreme to the furthest upstream position, and
+    then figure out the optimal x and y for each lens pack.
+    Move the motor to the other side, the most downstream side and do the same
+    thing. Save the x and y here as well.
+
+    So this focus motor, moves three motors in combination. The z is used for
+    the main calculations records, to get a focal spot size, and x
+    and y are moved based on the tweak that was done at the beginning.
+
+    In general, it saves an x, y, z at the ends of motion, the idea being that
+    the beam is a line that is not colinear with any one axis, so we draw a
+    line in 3D space to follow the center of the beam.
+
+    The scientist workflow is basically the same as before: you call one
+    function (align) with low beam to find the centers of the lenses, then you
+    have two pseudo motors (calib_z, beam_size) that magically put the lens in
+    the right spot.
+
+    Notes
+    -----
+    Use `pcdscalc.be_lens_cals.configure_defaults` function to set some default
+    parameters used in some calculations for different hutches.
+    Use `pcdscalc.set_lens_set_to_file` to set the lens sets file
+    """
     x = FCpt(IMS, '{self.x_prefix}')
     y = FCpt(IMS, '{self.y_prefix}')
     z = FCpt(IMS, '{self.z_prefix}')
 
     calib_z = Cpt(PseudoSingleInterface)
     beam_size = Cpt(PseudoSingleInterface)
 
-    tab_whitelist = ['tweak', 'align']
+    tab_whitelist = ['tweak', 'align', 'calib_z', 'beam_size', 'create_lens',
+                     'read_lens']
     tab_component_names = True
 
+    # TODO: do we want these guys to be None? it seems like if they
+    # are None calculations fail - at least z_offset and z_dir should
+    # probably not be None
```

Comment:
I agree. These were probably making some assumptions about how the underlying calculation functions handled a `None` input argument, and should be changed along with the change of the underlying calculation functions.

I suspect any of these that are configurations in pcdscalcs (beamsize_unfocused) should also be removed or reworked- note how if you make this object with default args you'll ignore the configured value for that parameter.

Reply:
Yeah I took the `beamsize_unfocused` out (in d3c2985)- because we can configure this default value through `pcdscalc.be_lens_calcs` - i think this was the only one that needed to be removed for now